core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -772,6 +822,45 @@ public boolean load(String graphHopperFolder) {
         }
     }
 
+    private void checkProfilesConsistency() {
+        for (ProfileConfig profile : profilesByName.values()) {
+            if (!encodingManager.hasEncoder(profile.getVehicle())) {
+                throw new IllegalArgumentException("Unknown vehicle '" + profile.getVehicle() + "' in profile: " + profile + ". Make sure to add all vehicle used in profiles to 'graph.flag_encoders'");
+            }
+            FlagEncoder encoder = encodingManager.getEncoder(profile.getVehicle());
+            if (profile.isTurnCosts() && !encoder.supportsTurnCosts()) {
+                throw new IllegalArgumentException("The profile '" + profile.getName() + "' was configured with " +
+                        "'turn_costs=true', but the corresponding vehicle '" + profile.getVehicle() + "' does not support turn costs." +
+                        "\nYou need to add `|turn_costs=true` to the vehicle in `graph.flag_encoders`");
+            }
+            try {
+                createWeighting(new HintsMap(profile.getWeighting()), encoder, NO_TURN_COST_PROVIDER);
```

Comment:
Sorry for being a bit lat to look at the changes. I was wondering why are we creating an instance of the weighting here? Wouldn't it be enough to check if a weighting with a certain name is "registered"/"available"? Or is this the easiest option to check if the weighting is available?

Reply:
Well yes the weighting is only created to check whether or not this is possible. Simply creating it seems like the easiest (and certainly most complete) way to do this. For example with custom weightings in #1776 we also get an error if something is wrong about the custom model YAML configuration. Then again like you observed there is no real *need* to create the weighting. Do you see a problem with this?

Reply:
> For example with custom weightings in #1776 we also get an error if something is wrong about the custom model YAML configuration this way.

Ah I see, that's a strong point to do this and makes sense.

> Then again, like you observed, there is no real need to create the weighting. Do you see a problem with this?

This created an exception for me. I use a weighting that requires certain parameters in the HintsMap, otherwise an exception is thrown. It is easy enough to fix, so that's no problem.